microsoft/testsuites/mshv/mshv_root_tests.py

```diff
@@ -1,223 +1,87 @@
-    category="stress",
+    category="functional",
     description="""
-    This test suite contains tests that are meant to be run on the
-    Microsoft Hypervisor (MSHV) root partition.
+    This test suite contains tests that should be run on the
+    Microsoft Hypervisor (MSHV) root partition. This test suite contains tests
+    to check health of mshv root node.
     """,
 )
 class MshvHostTestSuite(TestSuite):
-    IGVM_PATH_VARIABLE = "igvm_path"
-    CONFIG_VARIABLE = "mshv_vm_create_stress_configs"
-    DEFAULT_ITERS = 15
-    DEFAULT_CPUS_PER_VM = 1
-    DEFAULT_MEM_PER_VM_MB = 1024
-    DEFAULT_GUEST_VM_TYPE = "NON-CVM"
-
-    HYPERVISOR_FW_NAME = "hypervisor-fw"
-    DISK_IMG_NAME = "vm_disk_img.raw"
 
+    mshvdiag_dmesg_pattern = re.compile(r"\[\s+\d+.\d+\]\s+mshv_diag:.*$")
     def before_case(self, log: Logger, **kwargs: Any) -> None:
         node = kwargs["node"]
         if not node.tools[Ls].path_exists("/dev/mshv", sudo=True):
             raise SkippedException("This suite is for MSHV root partition only")
 
-        # add user to mshv group for access to /dev/mshv
-        node.tools[Usermod].add_user_to_group("mshv", sudo=True)
+        if not node.tools[Ls].path_exists("/dev/mshv_diag", sudo=True):
+            raise SkippedException("mshv_diag module should be loaded on MSHV"
+                                   " root parition.")
 
-        working_path = node.get_working_path()
-        node.tools[Wget].get(
-            "https://github.com/cloud-hypervisor/rust-hypervisor-firmware/releases/download/0.4.1/hypervisor-fw",  # noqa: E501
-            file_path=str(working_path),
-            filename=self.HYPERVISOR_FW_NAME,
-        )
-        node.tools[Wget].get(
-            "https://cloud-images.ubuntu.com/focal/current/focal-server-cloudimg-amd64.img",  # noqa: E501
-            file_path=str(working_path),
-            filename=f"{self.DISK_IMG_NAME}.img",
-        )
-        node.tools[QemuImg].convert(
-            "qcow2",
-            str(working_path / f"{self.DISK_IMG_NAME}.img"),
-            "raw",
-            str(working_path / self.DISK_IMG_NAME),
-        )
 
     @TestCaseMetadata(
         description="""
-        Stress the MSHV virt stack by repeatedly creating and destroying
-        multiple VMs in parallel. By default creates VMs with 1 vCPU and
-        1 GiB of RAM each. Number of VMs createdis equal to the number of
-        CPUs available on the host. By default, the test is repeated 25
-        times. All of these can be configured via the variable
-        "mshv_vm_create_stress_configs" in the runbook.
+        With mshv_diag module loaded, ensure mshvlog.service starts and runs
+        successfully on MSHV root partitions. Also confirm there are no errors
+        reported by mshv_diag module in dmesg.
         """,
         priority=4,
-        timeout=10800,  # 3 hours
+        timeout=30,  # 30 seconds
     )
-    def stress_mshv_vm_create(
+    def mshvlog_is_active(
```

Comment:
Test case name should start with verify_ if it's a functional test case.

Reply:
makes sense, I will fix it.

Reply:
done!